src/osp/Active/SysVehicle.cpp

```diff
@@ -331,8 +368,15 @@ ActiveEnt SysVehicle::part_instantiate(PrototypePart& part,
             collision.m_shape = cd.m_type;
 
         }
-    }
 
+        // Save the list of machines this object owns
+        MachineDef machineDef
+        {
+            currentEnt,
+            currentPrototype.m_machineIndices
+        };
+        machineMapping.push_back(std::move(machineDef));
```

Comment:
When the objects are instantiated, the PrototypeObject's machines are saved in this array for later instantiation